WooCommerce/Classes/Authentication/AuthenticationManager.swift

```diff
@@ -217,8 +219,31 @@ extension AuthenticationManager: WordPressAuthenticatorDelegate {
             return
         }
 
-        let isSelfHosted = site.isWP && !site.isWPCom
-        let authenticationResult: WordPressAuthenticatorResult = .presentPasswordController(value: isSelfHosted)
+        /// WordPress must be present.
+        guard site.isWP else {
+            let viewModel = NotWPErrorViewModel()
+            let notWPErrorUI = ULErrorViewController(viewModel: viewModel)
+
+            let authenticationResult: WordPressAuthenticatorResult = .injectViewController(value: notWPErrorUI)
+
+            onCompletion(authenticationResult)
+
+            return
+        }
+
+        /// For self-hosted sites, navigate to enter the email address associated to the wp.com account:
+        /// https://github.com/woocommerce/woocommerce-ios/issues/3426
+        guard site.isWPCom else {
+            let authenticationResult: WordPressAuthenticatorResult = .presentEmailController
+
+            onCompletion(authenticationResult)
+
+            return
+        }
+
+        /// We should never reach this point, as WPAuthenticator won't call its delegate for this case.
+        ///
+        let authenticationResult: WordPressAuthenticatorResult = .presentPasswordController(value: false)
```

Comment:
since this path is not expected to happen, does it make sense to return an error result and log an error like with `DDLogError`?

Reply:
Good point. WPAuthenticator should implement the case and not call into the delegate, but if for some reason it does, I think it would be better to implement it just in case. I am very scared of making assumptions about what libraries do, in particular WPAuthenticator.

Logging this case makes a lot of sense. I'll add a line here